app/controllers/assignments_controller.rb

```diff
@@ -44,7 +44,8 @@ def generate_rotation
       flash[:errors] = 'The starting user must be in the rotation.'
       # ... and return is correct here
       # rubocop:disable Style/AndOr
-      redirect_to :back and return
+      redirect_back(fallback_location:
+                    roster_assignments_path(@roster)) and return
```

Comment:
Shouldn't need quite so many parentheses - I think you can safely lose the outer set.